apps/devhub/views.py

```diff
@@ -1735,3 +1737,22 @@ def docs(request, doc_name=None):
 def search(request):
     query = request.GET.get('q', '')
     return render(request, 'devhub/devhub_search.html', {'query': query})
+
+
+def api_key_agreement(request):
+    next_step = reverse('devhub.api_key')
+
+    if request.method == 'POST':
+        request.user.update(read_dev_agreement=now())
+        return redirect(next_step)
+
+    if request.user.read_dev_agreement is None:
+        return render(request, 'devhub/api/agreement.html')
+    else:
+        response = redirect(next_step)
+        return response
```

Comment:
this still looks like unnecessary copy/paste to me. Why not just add the template as a parameter to `submit()`?

    def api_key_agreement(request):
        return submit(request, next_step=..., 
                      template='devhub/api/agreement.html')